### DISCUSSION TITLE
Solving OPF in a PV distribution network using different operating modes (APC, RPC, MPPT, OID)

### DISCUSSION THREAD

**Discussion by gaadis** [lanl-ansi/PowerModelsDistribution.jl#440] [Q&A]
I have PV data for a 34-bus distribution network and I am trying to formulate an objective function that minimizes the power in each of these four operating modes
1. Active Power Control/Curtailment (APC)
2. Reactive Power Control (RPC)
3. Maximum Power Point Tracking (MPPT)
4. Optimized Inverter Dispatch (OID) (Joint control of real and reactive power)

Are there any functions in julia or the PMD package that solves these problems? If not, can anyone please provide some guidance on how to formulate these objective functions. I am trying to solve the OPF with the Branch Flow Model (BFM).

Any help will be greatly appreciated.

**Comment by pseudocubic** [comment]
There are currently no features in PMD that will enable you to do these right out of the box, but because it looks like you primarily want to change the objective function, this can be straightforward. I have answered a question similar to this recently [here](https://github.com/lanl-ansi/PowerModelsDistribution.jl/discussions/436#discussioncomment-6061721), which can get you started. 

Since you are interested in Branch Flow OPF, you will want to start with [this function](https://github.com/lanl-ansi/PowerModelsDistribution.jl/discussions/436#discussioncomment-6061721) and replace the objective with your own functions to achieve what you want. 

You can look at the `variable_mc_` functions to find which variables your particular formulation uses, or you can build an OPF problem with `instantiate_mc_model` and use the `var(pm)` function to explore which variables are available.
